### DISCUSSION TITLE
Python Version

### DISCUSSION THREAD

**Discussion by chengxs1994** [SohamTilekar/vidiopy#1] [General]
Suggest removing some obsolete syntax based on newer Python versions.

**Comment by SohamTilekar** [comment]
Could You Give me Some Code where there is Problem.

**Comment by chengxs1994** [comment]
```
type Num = int | float
type NumOrNone = None | Num
```

I need change to 

```
Num = int | float
NumOrNone = None | Num
```

My python version is 3.11

**Comment by chengxs1994** [comment]
I made simple modifications locally to make compatibility with 3.11.
https://github.com/SohamTilekar/vidiopy/pull/2

**Reply by SohamTilekar** [reply]
Plz Do not Use the 3.11 use the 3.12 The goal of the VidioPy is to Keep it As Up to date as possible You Could Use the python env.

**Reply by SohamTilekar** [reply]
I am Working on the Composite Video Clip You Should Work on the vidiopy.editor like moviepy.editor and I already have the AudioClip.py Add the Support to it.

**Reply by SohamTilekar** [reply]
I am making the commit and push in some hour you should re fork the project in some time.

**Reply by SohamTilekar** [reply]
Thanks for working with me. I am Begainer

**Reply by SohamTilekar** [reply]
To github

**Reply by chengxs1994** [reply]
Okay, readme needs to indicate the Python version. As far as I know, most users are using versions 3.8-3.11.

**Reply by SohamTilekar** [reply]
Ok I will make it Compatible with the 3.8-3.11
Could You Update the Read me file for now.
I am Fixing Bug It will take some time to push and commit code.

**Reply by chengxs1994** [reply]
Consider merging this PR, which is compatible with 3.11.
https://github.com/SohamTilekar/vidiopy/pull/2

**Reply by chengxs1994** [reply]
At the same time, I submitted another PR as an example of how users can quickly get through it.
https://github.com/SohamTilekar/vidiopy/pull/3

**Reply by SohamTilekar** [reply]
I Cant Merge Because I Just Push And it Seams that is Incompatible.

**Reply by SohamTilekar** [reply]
I renamed the src to the Vidiopy I have some Relative import Errors in the VideoClip, AudioClip.
Could You Help Me Solve it.
